infrastructure/resource_group_level/modules/cosmosdb.bicep

```diff
@@ -130,390 +130,368 @@ resource submissionsDb 'Microsoft.DocumentDB/databaseAccounts/mongodbDatabases@2
+      resource: {
+        id: 'tfm-facilities'
+        shardKey: {
+          _id: 'Hash'
         }
-      ]
+        indexes: [
+          {
+            key: {
+              keys: [
+                '_id'
+              ]
+            }
+          }
+        ]
+      }
     }
   }
-}
-
-resource tfmFeedbackCollection 'Microsoft.DocumentDB/databaseAccounts/mongodbDatabases/collections@2023-04-15' = {
-  parent: submissionsDb
-  name: 'tfm-feedback'
-  properties: {
-    resource: {
-      id: 'tfm-feedback'
-      indexes: [
-        {
-          key: {
-            keys: [
-              '_id'
-            ]
+  {
+    name: 'tfm-feedback'
+    properties: {
+      resource: {
+        id: 'tfm-feedback'
+        indexes: [
+          {
+            key: {
+              keys: [
+                '_id'
+              ]
+            }
           }
-        }
-      ]
+        ]
+      }
     }
   }
-}
-
-resource tfmTeamsCollection 'Microsoft.DocumentDB/databaseAccounts/mongodbDatabases/collections@2023-04-15' = {
-  parent: submissionsDb
-  name: 'tfm-teams'
-  properties: {
-    resource: {
-      id: 'tfm-teams'
-      indexes: [
-        {
-          key: {
-            keys: [
-              '_id'
-            ]
+  {
+    name: 'tfm-teams'
+    properties: {
+      resource: {
+        id: 'tfm-teams'
+        indexes: [
+          {
+            key: {
+              keys: [
+                '_id'
+              ]
+            }
           }
-        }
-      ]
+        ]
+      }
     }
   }
-}
-
-resource tfmUsersCollection 'Microsoft.DocumentDB/databaseAccounts/mongodbDatabases/collections@2023-04-15' = {
-  parent: submissionsDb
-  name: 'tfm-users'
-  properties: {
-    resource: {
-      id: 'tfm-users'
-      indexes: [
-        {
-          key: {
-            keys: [
-              '_id'
-            ]
+  {
+    name: 'tfm-users'
+    properties: {
+      resource: {
+        id: 'tfm-users'
+        indexes: [
+          {
+            key: {
+              keys: [
+                '_id'
+              ]
+            }
           }
-        }
-      ]
+        ]
+      }
     }
   }
-}
-
-resource usersCollection 'Microsoft.DocumentDB/databaseAccounts/mongodbDatabases/collections@2023-04-15' = {
-  parent: submissionsDb
-  name: 'users'
-  properties: {
-    resource: {
-      id: 'users'
-      shardKey: {
-        _id: 'Hash'
-      }
-      indexes: [
-        {
-          key: {
-            keys: [
-              '_id'
-            ]
-          }
+  {
+    name: 'users'
+    properties: {
+      resource: {
+        id: 'users'
+        shardKey: {
+          _id: 'Hash'
         }
-      ]
+        indexes: [
+          {
+            key: {
+              keys: [
+                '_id'
+              ]
+            }
+          }
+        ]
+      }
     }
-  }
-}
+  } 
+]
+
+// We set a batch size because otherwise Azure tries to create all of the resources in parallel and we get 429 errors.
+@batchSize(4)
```

Comment:
Then create them in a loop here, using this handy batch size limit. 4 seemed reasonable and worked when I tested it.